jsonpickle/pickler.py

```diff
@@ -539,6 +540,7 @@ def _flatten_obj_instance(self, obj):
         has_getnewargs_ex = util.has_method(obj, '__getnewargs_ex__')
         has_getinitargs = util.has_method(obj, '__getinitargs__')
         has_reduce, has_reduce_ex = util.has_reduce(obj)
+        self.excludes = getattr(obj, '_jsonpickle_exclude') if hasattr(obj, '_jsonpickle_exclude') else []
```

Comment:
`self.excludes = set(getattr(obj, '_jsonpickle_exclude', ()))` lets us do without the if-else expression.

It also lets us use a `set` object internally so it has a chance of being faster than `list` membership checks.

Reply:
Why not just convert it to a set at instantiation time? ``_flatten_obj_instance`` is called very often, so I can see this being a big perf penalty. Alternatively, if we really care about performance, it could be a tuple for small tuples (<5 items), and a set for larger ones. If it's only being used in ``util.items()``, then the ``in`` check will work both ways, though if this route is taken then ``self.exclude`` should become ``self._exclude`` so the user doesn't mess with it after it's gone through jsonpickle's type coercion at instantiation. I'll try to have this measured more concretely by Wednesday though, I'd do it today but I have a midterm due at midnight today for a summer class :(

Reply:
I concur, let's defer the conversion until instantiation.

Likewise, it should be able to be specified at creation time as a parameter to `__init__`.

Reply:
> then self.exclude should become self._exclude so the user doesn't mess with it after it's gone through jsonpickle's type coercion at instantiation

The only reason I vibe with for keeping it as `self.exclude` is that it lets us do without needing to have methods for adding and removing entries post-construction since our API would be `obj.exclude.add('...')`, but YAGNI so `self._exclude` is better since we can start out private by default and only expose it later if we have a strong reason to do so.

So +1 for renaming to `self._exclude` for the initial version. Plus, we can specify it at construction so there's much less need for additional API or direct attribute access. We may never actually need it.

Reply:
Renamed to `self._excludes`